src/platform/ASR/BUILD.gn

```diff
@@ -72,6 +72,7 @@ static_library("ASR") {
 
   deps = [
     "${chip_root}/src/app:app_config",
+    "${chip_root}/src/app/common:ids",
```

Comment:
So did this use to include Attributes.h or whatnot and not need to declare deps on it at all for some reason?  Trying to figure out why all these `common:ids` bits suddenly had to be added.

Reply:
platforms include things inside common but relied that include paths just worked.
They are not needed anymore as I kept the common bits, however include-wise this is still correct so I kept it. It is just that gn seemingly cannot enforce it.

Reply:
This is current state for obvious includes

![image](https://github.com/user-attachments/assets/360c6804-e2ce-41ba-80e5-a5452b3587a2)

However I am quite sure I had added each for a reason even if ASR is not directly in the above list.

Reply:
Very likely due to this (separate include path is painful ...):

![image](https://github.com/user-attachments/assets/f77cff3c-1143-4945-a252-3e1f18a23d69)

Likely these are used by _everyone_